pymc/model_graph.py

```diff
@@ -234,20 +248,133 @@ def make_graph(self, var_names: Optional[Iterable[VarName]] = None, formatting:
 
         return graph
 
+    def make_networkx(
+        self, var_names: Optional[Iterable[VarName]] = None, formatting: str = "plain"
+    ):
+        """Make networkx Digraph of PyMC model
+
+        Returns
+        -------
+        networkx.Digraph
+        """
+        try:
+            import networkx
+        except ImportError:
+            raise ImportError(
+                "This function requires the python library networkx, along with binaries. "
+                "The easiest way to install all of this is by running\n\n"
+                "\tconda install networkx"
+            )
+        graphnetwork = networkx.DiGraph(name=self.model.name)
+        for plate_label, all_var_names in self.get_plates(var_names).items():
+            if plate_label:
+                # # must be preceded by 'cluster' to get a box around it
+                subgraphnetwork = networkx.DiGraph(
+                    name="cluster" + plate_label, label=plate_label
+                )
+                for var_name in all_var_names:
+                    self._make_node(
+                        var_name,
+                        subgraphnetwork,
+                        nx=True,
+                        cluster="cluster" + plate_label,
+                        formatting=formatting,
+                    )
+                for sgn in subgraphnetwork.nodes:
+                    networkx.set_node_attributes(
+                        subgraphnetwork,
+                        {sgn: {"labeljust": "r", "labelloc": "b", "style": "rounded"}},
+                    )
+                node_data = {
+                    e[0]: e[1]
+                    for e in graphnetwork.nodes(data=True)
+                    & subgraphnetwork.nodes(data=True)
+                }
+
+                graphnetwork = networkx.compose(graphnetwork, subgraphnetwork)
+                networkx.set_node_attributes(graphnetwork, node_data)
+                graphnetwork.graph["name"] = self.model.name
+            else:
+                for var_name in all_var_names:
+                    self._make_node(
+                        var_name, graphnetwork, nx=True, formatting=formatting
+                    )
+
+        for child, parents in self.make_compute_graph(var_names=var_names).items():
+            # parents is a set of rv names that preceed child rv nodes
+            for parent in parents:
+                graphnetwork.add_edge(parent.replace(":", "&"), child.replace(":", "&"))
+        return graphnetwork
+
+
+def model_to_networkx(
+    model=None,
+    *,
+    var_names: Optional[Iterable[VarName]] = None,
+    formatting: str = "plain",
+):
+    """Produce a networkx Digraph from a PyMC model.
+
+    Requires networkx, which may be installed most easily with
```

Comment:
see the comment about the graphviz docstring, you need to use similar spacing than it does. the `.. code-block` needs an empty line before and after to be rendered correctly, the section titles need spaces right above them...

Reply:
Right I think i've fixed that.  Thanks for bearing with me on this